hedera-node/src/main/java/com/hedera/services/txns/token/TokenBurnTransitionLogic.java

```diff
@@ -77,26 +75,12 @@ public void doStateTransition() {
 		final var op = txnCtx.accessor().getTxn().getTokenBurn();
 		final var grpcId = op.getToken();
 		final var targetId = Id.fromGrpcToken(grpcId);
-
-		/* --- Load the model objects --- */
-		final var token = tokenStore.loadToken(targetId);
-		final var treasuryRel = tokenStore.loadTokenRelationship(token, token.getTreasury());
-		final var ownershipTracker = new OwnershipTracker();
+		final var amount = op.getAmount();
+		final var serialNumbersList = op.getSerialNumbersList();
 
 		/* --- Do the business logic --- */
-		if (token.getType().equals(TokenType.FUNGIBLE_COMMON)) {
-			token.burn(treasuryRel, op.getAmount());
-		} else {
-			final var burnList = op.getSerialNumbersList();
-			tokenStore.loadUniqueTokens(token, burnList);
-			token.burn(ownershipTracker, treasuryRel, burnList);
-		}
-
-		/* --- Persist the updated models --- */
-		tokenStore.commitToken(token);
-		tokenStore.commitTokenRelationships(List.of(treasuryRel));
-		tokenStore.commitTrackers(ownershipTracker);
-		accountStore.commitAccount(token.getTreasury());
+		final var burnLogic = new TokenBurnLogic();
```

Comment:
This should be injected into the constructor as we have it injected in the `Token Mint PR`